### PR TITLE
add keyboard shortcut and menu item to edit vector masks on layers

### PR DESCRIPTION
**Pull request by pineapplespatula**
has accompanying PR https://github.com/adobe-photoshop/spaces-adapter/pull/127

### CODE REVIEW

File: src/js/actions/layers.js [hunk 1]
```diff
@@ -1489,6 +1491,23 @@ define(function (require, exports) {
...
+        return this.transfer(tools.select, superselectVector)
+            .then(function () {
+                return descriptor.playObject(vectorMaskLib.editVectorMask());
+            }).then(function () {
+                return descriptor.playObject(vectorMaskLib.activateVectorMaskEditing());
+            });
+    };
+    editVectorMask.reads = [];
+    editVectorMask.writes = [locks.PS_DOC, locks.JS_DOC, locks.PS_APP, locks.JS_APP,
+    locks.PS_TOOL, locks.JS_TOOL, locks.PS_MENU, locks.JS_MENU, locks.JS_DIALOG,
```
Comment (pineapplespatula) [inline src/js/actions/layers.js]: indent here

File: src/js/actions/layers.js [hunk 2]
```diff
@@ -1489,6 +1491,23 @@ define(function (require, exports) {
     revealLayers.reads = [];
     revealLayers.writes = [locks.PS_DOC, locks.JS_DOC];
 
+
```
Comment (pineapplespatula) [inline src/js/actions/layers.js]: add JSDoc

File: src/js/actions/layers.js [hunk 3]
```diff
@@ -1490,6 +1492,29 @@ define(function (require, exports) {
...
+     * @return {Promise}
+     */
+    var editVectorMask = function () {
+        var toolStore = this.flux.store("tool"),
+            superselectVector = toolStore.getToolByID("superselectVector");
+
+        return this.transfer(tools.select, superselectVector)
+            .then(function () {
+                return descriptor.playObject(vectorMaskLib.editVectorMask());
+            }).then(function () {
```
Comment (iwehrman) [inline src/js/actions/layers.js]: The `.then` should be on a new line here.

File: src/js/actions/layers.js [hunk 4]
```diff
@@ -1490,6 +1492,29 @@ define(function (require, exports) {
...
+                return descriptor.playObject(vectorMaskLib.activateVectorMaskEditing());
```
Comment (iwehrman) [inline src/js/actions/layers.js]: Also, can you leave a note above each playObject call that summarizes what each one does? They have confusingly similar names...

**Comment by iwehrman** [comment]
This looks pretty good, but when I enter vector mask edit mode, I can't seem to do anything except move the mask around? In particular, I can't seem to resize the mask. Maybe that's because I created the vector mask from a path?

**Comment by pineapplespatula** [comment]
All vector masks are Paths, Currently we're just opening the Superselect vector tools. I can talk to seth about what else we want to expose here,

**Comment by pineapplespatula** [comment]
I talked to seth about next steps, and made a trello card. now the iterating begins

**Comment by iwehrman** [comment]
OK, sounds good. Would you mind giving us the summary in the standup tomorrow?

**Pull request merged**